datalake/src/main/java/com/sequenceiq/datalake/entity/DatalakeStatusEnum.java

```diff
@@ -97,6 +98,26 @@ public DatalakeStatusEnum mapToFailedIfInProgress() {
         }
     }
 
+    @SuppressWarnings("checkstyle:CyclomaticComplexity")
+    public boolean isFailedState() {
```

Comment:
If someone adds a new failed state he will have a chance to not add the state to this method. Can you please add a unit test which fails if this method doesn't return true for a `DatalakeStatusEnum` with __FAILED__ suffix in it? This might not be a perfect solution but at least forces the developer to deal with it.